.gitignore

```diff
@@ -1,3 +1,4 @@
+.idea
```

Comment:
rather than stick the editor's files in the repo, could you add a local `.gitignore_global`

```
❯ cat -p ~/.gitignore_global
.idea
.vscode
.env
```

Reply:
Hmm, I understand exactly why you'd want to push that into the .gitignore_global file, but doesn't that rely on all potential contributors having done the global configuration locally to exclude their favorite IDE's files? You can prevent these files/cruft creeping into future reviews, etc by just including it in the project's .gitignore file. I suppose it depends on whether your aim is to keep it out of the repository or to educate users that they should use the global ignore file, but it seems to me that there's reduced cognitive load in future reviews by including it.

Reply:
This is just my opinion @wolsen, but an editor/os is someone's personal choice -- not the responsibility of the repository

If Alice uses JetBrains,  she has an `.idea` folder in every repo to which she commits 
If Bob uses a Mac, his OS dumps a bunch of `.DS_Store` and `._*` files
If Catherine uses vscode, she's got a .vscode folder in every repo. 

You could argue that `git` should just KNOW these things exist and shouldn't commit them to the repo.

It seems to me a developer's personal environment shouldn't invade the repo's `.gitignore`.  I was hoping to mention the clean way to prevent a developer from adding `.idea` to every repo he or she commits to is to just update their own `~/.gitignore_global`

Reply:
I think its a fine opinion @addyess, and I happen to share the opinion that the `.gitignore_global` file is actually a better place for ignoring personal opinions that any repository has no business having opinions on. I also don't think projects need to have an inkling as to which editor I'm actually using.

I also just acknowledge that including these types of files in a `.gitignore` reduces review effort, potential cruft, and less than ideal experiences from potential contributors who _don't_ follow this (and likely without any ill intentions). Personally, I've typically just included these files in the .gitignore in the past as it was just easier for future me but I appreciate your willingness to educate here.

I do wonder if there'd be any merit in including a comment in the `.gitignore` file (not in this PR) indicating that IDE specific files do not go here, IDE specific files belong in your `~/.gitignore_global`. Saves cognitive load, educates contributors...

Reply:
@wolson I think i hear you suggesting this

```suggestion
# ideally, a developer would add this to their own `~/.gitignore_global`
.idea

```

Reply:
but there are .gitignore templates and in most of cases you have IDEs there. What is the problem?
you have project specific exclusions in the top of the file, then you have "this was automatically generated by..."

I am in favor of excluding IDEs. Example, official CPython: https://github.com/python/cpython/blob/924a3bfa28578802eb9ca77a66fb5d4762a62f14/.gitignore#L32